includes/db_api.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+include_once 'config.php';
+
+function registerUser($name, $username, $password, $user_code, $date_creation, $points, $isAdmin = false) {
+    global $conn;
+
+
+    //parâmetros da função 
+    $name = mysqli_real_escape_string($conn, $name); //é usado em todos os inputs de strings para prevenir sql injections
+    $username = mysqli_real_escape_string($conn, $username);
+    $password_hash = password_hash($password, PASSWORD_DEFAULT); // hashing
+    $user_code = mysqli_real_escape_string($conn, $user_code);
+    $points = mysqli_real_escape_string($conn, $points);
+    $date_creation = mysqli_real_escape_string($conn, $date_creation);
+
+    
+    $role_id_role = $isAdmin ? 0 : 1;
+
+    //verifica se user já existe
+    $query_check_username = "SELECT COUNT(*) as count FROM users WHERE username = '$username'"; 
```

Comment:
Sugestão: se só queres saber se há registos não precisas de verificar se a linha toda (`*`) existe ou não. Contar `0` ou outra constante qualquer é potencialmente mais rápido e dá na mesma.

```suggestion
    $query_check_username = "SELECT COUNT(0) as count FROM users WHERE username = '$username'"; 
```